contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -719,12 +752,21 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
             _wethSharePct < allowedWethShareStart ||
             _wethSharePct > allowedWethShareEnd
         ) {
+            if (dryRun) {
+                return false;
+            }
+
             revert PoolRebalanceOutOfBounds(
                 _wethSharePct,
                 allowedWethShareStart,
                 allowedWethShareEnd
             );
         }
+
+        // return before emitting the event is intentional
+        if (dryRun) {
+            return true;
+        }
         emit PoolRebalanced(_wethSharePct);
```

Comment:
Should we rename this event? It's emitted even when we don't do a rebalance.
I'm also starting to think if we really need this event here. It looks like it serves no purpose, this function doesn't even change state. We get the `_wethSharePct` when the tx reverts from the custom error above. We are not using this at all anywhere

Reply:
It is true that this event is emitted regardless of the function changing the active trading price of the pool (WETH share percentage in other words) or not. Meaning when we "rebalance" the WETH share from 14% to 16% the event will be emitted and I think it is a useful information to have.

Reply:
I don't have strong opinions about this but it still out of place since it's outside `_rebalance`. Not a big deal though

Reply:
I'm also wondering why can't we have a `getWETHShare` public function? That'll help us get the current WETH share in the active tick and would be more apt than an event since we can't even tracking this event in monitoring (it'd only emit when there's a rebalance not when there's a swap or something)

Reply:
I think you have a good point. I've refactored the WETH share into a separate function and emitted the event in the `_rebalance` function which is a more appropriate place for it: https://github.com/OriginProtocol/origin-dollar/pull/2266/commits/c7346e63a48e9658dc1973327ff93e3b7f959998